Modules/Authentication/Multi/MultiConcern.cs

```diff
@@ -0,0 +1,51 @@
+﻿using GenHTTP.Api.Content;
+using GenHTTP.Api.Protocol;
+
+namespace GenHTTP.Modules.Authentication.Multi;
+
+public sealed class MultiConcern : IConcern
+{
+    #region Get-/Setters
+
+    public IHandler Content { get; }
+
+    private readonly IConcern[] _delegatingConcerns;
+
+    #endregion
+
+    #region Initialization
+
+    public MultiConcern(IHandler content, IConcern[] delegatingConcerns)
+    {
+        Content = content;
+        _delegatingConcerns = delegatingConcerns ?? [];
+    }
+
+    #endregion
+
+    #region Functionality
+
+    public async ValueTask<IResponse?> HandleAsync(IRequest request)
+    {
+        IResponse? lastResponse = null;
+        foreach (var concern in _delegatingConcerns)
+        {
+            lastResponse = await concern.HandleAsync(request);
+
+            if (lastResponse?.Status.KnownStatus != ResponseStatus.Unauthorized)
+            {
+                return lastResponse;
+            }
+        }
+
+        return lastResponse ??
+               request.Respond()
```

Comment:
I'm not super happy about this. This only happens if the delegating concerns array is empty. Which strictly speaking makes no sense. It would be perhaps better to throw error during construction that no underlying auth methods were provided? What is your opinion?

Reply:
Yes, I would throw an exception during `Build()` - but this place will probably stay the same anyway because you do not want to do a `First()` here 🤔